source/javascripts/components/StepDrawer/StepDrawer.tsx

```diff
@@ -0,0 +1,37 @@
+import { Box, Drawer, Text, useDisclosure } from '@bitrise/bitkit';
+import { FormProvider, useForm } from 'react-hook-form';
+import { FilterFormValues } from './StepDrawer.types';
+import StepFilter from './components/StepFilter';
+import StepList from './components/StepList';
+import { useCategories, useFilter } from './hooks/useSteps';
+
+type Props = {
+  isOpen: boolean;
+};
+
+const StepDrawer = ({ isOpen: isInitialOpen }: Props) => {
+  const { isOpen, onClose } = useDisclosure({ defaultIsOpen: isInitialOpen });
+  const { categories } = useCategories();
+  const form = useForm<FilterFormValues>({
+    defaultValues: {
+      search: '',
+      categories: [],
+    },
+  });
+
+  const { steps } = useFilter(form.watch());
```

Comment:
I think it's more performant to put this into the StepList component because that way it avoids re-rendering the whole drawer component on change.